Docs: Adding how to use plugin version, through docker env variable

joachimlindborg opened the pull request:
It was unclear if one can specify the plugin version to the environment variable when installing plugins when instatiating the docker componen. It is possible and works fine. 
Added some text to the documentation

Tested towards grafana 5.3.4

<!--

Thank you for sending a pull request! Here are some tips:

1. If this is your first time, please read our contribution guide at https://github.com/grafana/grafana/blob/master/CONTRIBUTING.md

2. Ensure you include and run the appropriate tests as part of your Pull Request.

3. In a new feature or configuration option, an update to the documentation is necessary. Everything related to the documentation is under the docs folder in the root of the repository.

4. If the Pull Request is a work in progress, make use of GitHub's "Draft PR" feature and mark it as such.

5. If you can not merge your Pull Request due to a merge conflict, Rebase it. This gets it in sync with the master branch.

6. Name your PR as "<FeatureArea>: Describe your change", e.g. Alerting: Prevent race condition. If it's a fix or feature relevant for the changelog describe the user impact in the title. The PR title is used to auto-generate the changelog for issues marked with the "add to changelog" label.

-->

**This PR clarifies documentation / love your work**:

**fixes documentation*:

<!--

* Automatically closes linked issue when the Pull Request is merged.

Usage: "Fixes #<issue number>", or "Fixes (paste link of issue)"

-->

Fixes #

**Special notes for your reviewer**:

joachimlindborg committed:
Adding grafan plugin version info

It was unclear if one can specify the plugin version to the environment variable. It is possible and works fine. Added it to the plugin information.

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/grafana/grafana?pullRequest=19924) <br/>Thank you for your submission, we really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/grafana/grafana?pullRequest=19924) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/grafana/grafana?pullRequest=19924) it.</sub>

olivierlemasle commented on docs/sources/installation/docker.md:
```diff
@@ -74,7 +74,7 @@ For a list of available tags, check out [grafana/grafana](https://hub.docker.com
 ## Installing Plugins for Grafana
 
-Pass the plugins you want installed to docker with the `GF_INSTALL_PLUGINS` environment variable as a comma separated list. This will pass each plugin name to `grafana-cli plugins install ${plugin}` and install them when Grafana starts.
+Pass the plugins you want installed to docker with the `GF_INSTALL_PLUGINS` environment variable as a comma separated list. This will pass each plugin name to `grafana-cli plugins install ${plugin}` and install them when Grafana starts. If you need to specify a specific plugin version you can add optional version to plugin evnironment variable otherwise latest will be assumed `--build-arg "GF_INS ...
```
There's a typo in "environment".

joachimlindborg replied:
```suggestion
Pass the plugins you want installed to docker with the `GF_INSTALL_PLUGINS` environment variable as a comma separated list. This will pass each plugin name to `grafana-cli plugins install ${plugin}` and install them when Grafana starts. If you need to specify a specific plugin version you can add optional version to plugin environment variable otherwise latest will be assumed `--build-arg "GF_INSTALL_PLUGINS=grafana-clock-panel 1.0.1,grafana-simple-json-datasource 1.3.5"`
```

olivierlemasle commented on docs/sources/installation/docker.md:
```diff
@@ -87,7 +87,7 @@ docker run \
 ## Building a custom Grafana image with pre-installed plugins
 
-In the [grafana-docker](https://github.com/grafana/grafana/tree/master/packaging/docker)  there is a folder called `custom/` which includes a `Dockerfile` that can be used to build a custom Grafana image.  It accepts `GRAFANA_VERSION` and `GF_INSTALL_PLUGINS` as build arguments.
+In the [grafana-docker](https://github.com/grafana/grafana/tree/master/packaging/docker)  there is a folder called `custom/` which includes a `Dockerfile` that can be used to build a custom Grafana image.  It accepts `GRAFANA_VERSION` and `GF_INSTALL_PLUGINS` as build arguments. If you need to specify a specific plugin version you can add version to plugin evnironment variable `--build-arg "GF_IN ...
```
same typo here.

joachimlindborg replied:
hope I fixed it right now

papagian replied:
The typo is not fixed.

joachimlindborg committed:
Update typo docs/sources/installation/docker.md

papagian commented on docs/sources/installation/docker.md:
```diff
@@ -74,7 +74,7 @@ For a list of available tags, check out [grafana/grafana](https://hub.docker.com
 ## Installing Plugins for Grafana
 
-Pass the plugins you want installed to docker with the `GF_INSTALL_PLUGINS` environment variable as a comma separated list. This will pass each plugin name to `grafana-cli plugins install ${plugin}` and install them when Grafana starts.
+Pass the plugins you want installed to docker with the `GF_INSTALL_PLUGINS` environment variable as a comma separated list. This will pass each plugin name to `grafana-cli plugins install ${plugin}` and install them when Grafana starts. If you need to specify a specific plugin version you can add optional version to plugin environment variable otherwise latest will be assumed `--build-arg "GF_INS ...
```
There is a missing `.` at the end of the sentence.
Also, instead of `--build-arg` should be `-e` since the example below is a `docker run` instead of a `docker build` command.
Finally, I would propose to move the new sentence under the example `docker run` so the reader will have a better understanding of the whole context.

papagian reviewed (changes requested):
Thank you a lot for contributing this.
Please check my comments for some some modifications I have asked.

joachimlindborg committed:
Applied all comments

Moved the information to make it more readable, fixed typos and added missing punctuation.

papagian commented on docs/sources/installation/docker.md:
````diff
@@ -85,6 +85,8 @@ docker run \
   grafana/grafana
 ```
 
+> If you need to specify the version of a plugin, you can add it to the command. Otherwise latest will be assumed. `-e "GF_INSTALL_PLUGINS=grafana-clock-panel 1.0.1,grafana-simple-json-datasource 1.3.5"`
````
Great, I liked the idea that you have used notes!
I also preferred it as it was before the last modification, so I would change it to:
> If you need to specify the version of a plugin, you can add it to the `GF_INSTALL_PLUGINS` environment variable. Otherwise, the latest will be assumed.
For example: `-e "GF_INSTALL_PLUGINS=grafana-clock-panel 1.0.1,grafana-simple-json-datasource 1.3.5"`

papagian commented on docs/sources/installation/docker.md:
````diff
@@ -103,6 +105,8 @@ docker run \
   grafana:latest-with-plugins
 ```
 
+> If you need to specify the version of a plugin, you can add it to the build argument. Otherwise latest will be assumed. `--build-arg "GF_INSTALL_PLUGINS=grafana-clock-panel 1.0.1,grafana-simple-json-datasource 1.3.5"`
````
Same here, I would change it to:
> If you need to specify the version of a plugin, you can add it to the `GF_INSTALL_PLUGINS` build argument. Otherwise, the latest will be assumed.
For example: `--build-arg "GF_INSTALL_PLUGINS=grafana-clock-panel 1.0.1,grafana-simple-json-datasource 1.3.5"`

papagian reviewed (changes requested):
Thank you for making the requested modifications.
I would recommend some minor changes before finally merging it.

joachimlindborg committed:
Rewording and better example

More feedback, added example text. Did not manage to get newline for the example.

papagian commented on docs/sources/installation/docker.md:
````diff
@@ -85,6 +85,8 @@ docker run \
 ```
 
+> If you need to specify the version of a plugin, you can add it to the `GF_INSTALL_PLUGINS` environment variable. Otherwise, the latest will be assumed. For example:`-e "GF_INSTALL_PLUGINS=grafana-clock-panel 1.0.1,grafana-simple-json-datasource 1.3.5"`
+
````
I think that it would be easier to read if you added a space after `For example:`

papagian reviewed (changes requested):
Great, thanks!
I have asked for a minor modification before merging it.

joachimlindborg committed:
added the missing space

papagian reviewed (approved):
Great, thank you for your contribution.
